majortom_gateway/gateway_api.py

```diff
@@ -79,9 +79,14 @@ async def connect(self):
                 ssl_context.verify_mode = ssl.CERT_NONE
 
         logger.info("Connecting to Major Tom")
-        self.websocket = await websockets.connect(self.gateway_endpoint,
-                                                  extra_headers=self.headers,
-                                                  ssl=ssl_context)
+        while True:
+            self.websocket = await websockets.connect(self.gateway_endpoint,
+                                                      extra_headers=self.headers,
+                                                      ssl=ssl_context)
+            if self.websocket:
+                break
+            logger.warn("  Connection attempt didn't produce a websocket object")
+            await asyncio.sleep(0.1)
```

Comment:
An alternative would be to raise a specific error and rely on the `connect_with_retries()` method to catch it.

Reply:
Which do you prefer?  Which is more idiomatic for Python?

Reply:
I think I answered this in Slack, but I would prefer having all retry logic contained in `connect_with_retries()`. So if the connection fails, just blast out an error and catch it on the other side.